roles/ssh_hardening/README.md

```diff
@@ -1,257 +1,437 @@
+![devsec.ssh_hardening](https://github.com/dev-sec/ansible-collection-hardening/workflows/devsec.ssh_hardening/badge.svg)
 
 ## Description
 
 This role provides secure ssh-client and ssh-server configurations. It is intended to be compliant with the [DevSec SSH Baseline](https://github.com/dev-sec/ssh-baseline).
 
 Warning: This role disables root-login on the target server! Please make sure you have another user with su or sudo permissions that can login into the server.
 
-## Requirements
+<!-- BEGIN_ANSIBLE_DOCS -->
+## Supported Operating Systems
+| Platform | Versions |
+| -------- | -------- |
+| EL | 7, 8, 9 |
+| Ubuntu | bionic, focal, jammy |
+| Debian | buster, bullseye |
+| Amazon |  |
+| Fedora |  |
+| ArchLinux |  |
+| SmartOS |  |
+| FreeBSD | 12.2 |
+| OpenBSD | 7.0 |
 
-- Ansible >= 2.9
-- root-privileges on the target system
+## Role Variables
 
-As this role requires root-privileges, we added `become: true` to all tasks. So please make sure you run the role as root or as a user with become-privileges.
+* `network_ipv6_enable`
+  * Default: `true`
+  * Description: false if IPv6 is not needed. `ssh_listen_to` must also be set to listen to IPv6 addresses (for example `[::]`).
+  * Type: bool
+  * Required: no
+* `ssh_client_config_file`
+  * Default: `/etc/ssh/ssh_config`
+  * Description: path of the ssh client configuration file, e.g. `/etc/ssh/ssh_config.d/custom.conf`.
+  * Type: str
+  * Required: no
+* `ssh_server_config_file`
+  * Default: `/etc/ssh/sshd_config`
+  * Description: path of the ssh server configuration file, e.g. `/etc/ssh/sshd_config.d/custom.conf`.
+  * Type: str
+  * Required: no
+* `ssh_server_ports`
+  * Default: `["22"]`
+  * Description: ports on which ssh-server should listen.
+  * Type: list
+  * Required: no
+* `ssh_client_port`
+  * Default: `22`
+  * Description: Specifies the port number to connect on the remote host.
+  * Type: str
+  * Required: no
+* `ssh_listen_to`
+  * Default: `["0.0.0.0"]`
+  * Description: one or more ip addresses, to which ssh-server should listen to. Default i all IPv4 addresses, but should be configured to specific addresses for security reasons
+  * Type: list
+  * Required: no
+* `ssh_host_key_files`
+  * Default: ``
+  * Description: Host keys for sshd. If empty ['/etc/ssh/ssh_host_rsa_key', '/etc/ssh/ssh_host_ecdsa_key', '/etc/ssh/ssh_host_ed25519_key'] will be used, as far as supported by the installed sshd version.
+  * Type: list
+  * Required: no
+* `ssh_host_rsa_key_size`
+  * Default: `4096`
+  * Description: Specifies the number of bits in the private host RSA key to create.
+  * Type: str
+  * Required: no
+* `ssh_host_key_algorithms`
+  * Default: ``
+  * Description: Host key algorithms that the server offers. If empty the default lst will be used. Otherwise overrides the setting with specified list of algorithms. Check `man sshd_config`, `ssh -Q HostKeyAlgorithms` or other sources for supported algorithms - make sure you check the correct version
```

Comment:
```suggestion
  * Description: Host key algorithms that the server offers. If empty the default list will be used. Otherwise overrides the setting with specified list of algorithms. Check `man sshd_config`, `ssh -Q HostKeyAlgorithms` or other sources for supported algorithms - make sure you check the correct version
```